src/models/Articles.js

```diff
@@ -376,4 +377,42 @@ export default class Articles extends Model {
 
     return similarity >= 0.6;
   }
+
+  /** Get one article by slug
+   *
+   * @static
+   * @memberof Articles
+   *
+   * @param {string} slug - slug of the article to be sort
+   * @param {Object} options - optional object for database query option
+   *
+   * @returns {Array} - array
+   */
+  static async findOneArticle(slug, options = {}) {
+    const article = await Articles.findOne({
+      where: {
+        slug: { [Sequelize.Op.iLike]: slug }, ...options
+      }
+    });
+    return article;
+  }
+
+  /** Update an article by slug
+   *
+   * @static
+   * @memberof Articles
+   *
+   * @param {Object} article - article object
+   * @param {Object} slug - slug of the article
+   * @returns {Array} - array
+   */
+  static async updateArticle(article, slug) {
+    article = await Articles.update(article, {
```

Comment:
Can we send the article instance that was gotten in the isAuthor method here so that we don't have to do a "where" again. We just do `article.update(...)`. So in the `isAuthor` middleware,  we can do 
```javascript
req.articleInstance = article;
```
And then pass in the article instance here.

Reply:
We can skip this if it doesn't let us return the updated field though. I just want us to keep the db queries to a minimum.